### PR TITLE
[SPARK-47930][BUILD] Upgrade RoaringBitmap to 1.0.6

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `RoaringBitmap` from `1.0.5` to `1.0.6`.

### Why are the changes needed?
The full release notes:
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/1.0.6

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
The benchmark(org.apache.spark.MapStatusesConvertBenchmark) results as follows:
JDK 17:
JDK 21:

File: core/benchmarks/MapStatusesConvertBenchmark-results.txt [hunk 1]
```diff
@@ -2,12 +2,12 @@
 MapStatuses Convert Benchmark
 ================================================================================================
 
-OpenJDK 64-Bit Server VM 17.0.10+7-LTS on Linux 6.5.0-1016-azure
+OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1018-azure
```
Comment (panbingkun) [inline core/benchmarks/MapStatusesConvertBenchmark-results.txt:5]: The test results of the version `1.0.6` based on `JDK17` are better than the version `1.0.5`, but it seems that the test results based on `JDK21` are slightly worse.

**Comment by panbingkun** [comment]
cc @dongjoon-hyun @LuciferYang

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @panbingkun .

**Comment by panbingkun** [comment]
According to @LuciferYang's suggestion, I ran the benchmark test (based on JDK21)  again. 
This time, the performance was quite good, as follows:
https://github.com/panbingkun/spark/actions/runs/8780180191
image
@dongjoon-hyun @LuciferYang Thank you very much ❤️ !

**Pull request closed without merging**